src/transformers/models/falcon/modeling_falcon.py

```diff
@@ -99,19 +99,23 @@ def _set_cos_sin_cache(self, seq_len, device, dtype):
         self.cos_cached = self.cos_cached.type(dtype)
         self.sin_cached = self.sin_cached.type(dtype)
 
-    def cos_sin(self, seq_len: int, past_key_values_length: int, device="cpu", dtype=torch.bfloat16) -> torch.Tensor:
+    def cos_sin(self, seq_len: int, past_key_values_length: int, position_ids: torch.Tensor, device="cpu", dtype=torch.bfloat16) -> torch.Tensor:
         total_length = seq_len + past_key_values_length
         if total_length > self.seq_len_cached:
             self._set_cos_sin_cache(total_length, device, dtype)
-        return (
-            self.cos_cached[:, past_key_values_length : seq_len + past_key_values_length],
```

Comment:
the slicing here is equivalent to building position ids from the sequence length, without taking into account any potential left-padding